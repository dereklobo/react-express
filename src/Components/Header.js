const Header = () => {
  return (
    <div className="bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500">
    
    <div className="container mx-auto max-w-5xl px-3 py-6">
      <p className="block font-bold text-xl font-opensans">Nasa <span className="text-primary">Images</span></p>
      <h1>Header</h1>
      <div className="my-14">
        <h2 className="font-bold text-xl max-w-xs mb-2">Mars Rover Photos</h2>
        <p className="font-light opacity-80 hidden md:inline-block">Photos captured by rover cameras</p>

        {/* {Form} */}
        
      </div>
    </div>
    
    </div>
  )
}

export default Header