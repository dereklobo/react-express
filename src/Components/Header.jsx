import { useState } from "react"

const Header = ({data,setFilterData, isLoading}) => {
  const [inputValue, setInputValue] = useState('');
  const filterPhotosByCameraIdOrName = (data) =>  data && data.length ? data.filter((item) => 
    { 
      return item.id.toString().includes(inputValue) 
    } 
  )
      : [] 
  
  const handleInputChange = (e) => setInputValue(e.target.value)
  const handleSubmit = (e) => {
    e.preventDefault()

    if(inputValue!=='') {
      setFilterData(filterPhotosByCameraIdOrName(data))
    } else {
      setFilterData(data)
    }
  }

  return (
    <div className="bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500">
    
    <div className="container mx-auto max-w-5xl px-3 py-6">
      <p className="block font-bold text-xl font-opensans">Nasa <span className="text-primary">Images</span></p>
      <h1>Header</h1>
      <div className="my-14">
        <h2 className="font-bold text-xl max-w-xs mb-2">Mars Rover Photos</h2>
        <p className="font-light opacity-80 hidden md:inline-block">Photos captured by rover cameras</p>

        {/* {Form} */}
        <form className="relative my-8 max-w-md" onSubmit={handleSubmit}>
          <input onInput={handleInputChange} value={inputValue} className="indent-2 py-4 pr-[4.5rem] rounded-sm bg-gray-50
          w-full md:max-w-md focus:ring-primary outline-none focus:ring-2
          " placeholder="Search camera id, camera name" type="text"/>

          <button type="submit" className="absolute bottom-3.5 right-2.5 bg-primary text-white
          p-1 rounded-sm text-sm">
            <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth={1.5} stroke="currentColor" className="size-6">
              <path strokeLinecap="round" strokeLinejoin="round" d="M17.25 8.25 21 12m0 0-3.75 3.75M21 12H3" />
            </svg>

          </button>
        </form>
      </div>
    </div>
    
    </div>
  )
}

export default Header