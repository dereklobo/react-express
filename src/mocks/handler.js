// src/mocks/handlers.js
import { http, HttpResponse } from 'msw'
 
export const handlers = [
  // Intercept "GET https://example.com/user" requests...
  http.get('http://localhost:9000/nasa', () => {
    // ...and respond to them using this JSON response.
    return HttpResponse.json(
        {
            "id": 2096,
            "sol": 2,
            "camera": {
              "id": 26,
              "name": "NAVCAM",
              "rover_id": 5,
              "full_name": "Navigation Camera"
            },
            "img_src": "http://mars.jpl.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/00002/opgs/edr/ncam/NLA_397671934EDR_F0010008AUT_04096M_.JPG",
            "earth_date": "2012-08-08",
            "rover": {
              "id": 5,
              "name": "Curiosity",
              "landing_date": "2012-08-06",
              "launch_date": "2011-11-26",
              "status": "active",
              "max_sol": 4102,
              "max_date": "2024-02-19",
              "total_photos": 695670,
              "cameras": [
                {
                  "name": "FHAZ",
                  "full_name": "Front Hazard Avoidance Camera"
                },
                {
                  "name": "NAVCAM",
                  "full_name": "Navigation Camera"
                },
                {
                  "name": "MAST",
                  "full_name": "Mast Camera"
                },
                {
                  "name": "CHEMCAM",
                  "full_name": "Chemistry and Camera Complex"
                },
                {
                  "name": "MAHLI",
                  "full_name": "Mars Hand Lens Imager"
                },
                {
                  "name": "MARDI",
                  "full_name": "Mars Descent Imager"
                },
                {
                  "name": "RHAZ",
                  "full_name": "Rear Hazard Avoidance Camera"
                }
              ]
            }
          })
  }),
]