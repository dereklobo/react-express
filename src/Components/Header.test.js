import { render, screen } from '@testing-library/react';
import Header from './Header';
import { setupServer } from 'msw/browser';
import { handlers } from '../mocks/handler'
 
export const server = setupServer(...handlers)

test('renders learn react link', () => {
  const { container } = render(<Header />);
  const header = screen.getByText(/Header/i);
  expect(container).toMatchSnapshot()
  expect(header).toBeInTheDocument();
});
