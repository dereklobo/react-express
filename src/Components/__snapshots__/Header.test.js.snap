// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <div
    class="bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500"
  >
    <div
      class="container mx-auto max-w-5xl px-3 py-6"
    >
      <p
        class="block font-bold text-xl font-opensans"
      >
        Nasa 
        <span
          class="text-primary"
        >
          Images
        </span>
      </p>
      <h1>
        Header
      </h1>
      <div
        class="my-14"
      >
        <h2
          class="font-bold text-xl max-w-xs mb-2"
        >
          Mars Rover Photos
        </h2>
        <p
          class="font-light opacity-80 hidden md:inline-block"
        >
          Photos captured by rover cameras
        </p>
        <form
          class="relative my-8 max-w-md"
        >
          <input
            class="indent-2 py-4 pr-[4.5rem] rounded-sm bg-gray-50 w-full md:max-w-md focus:ring-primary outline-none focus:ring-2 "
            placeholder="Search camera id"
            type="text"
            value=""
          />
          <button
            class="absolute bottom-3.5 right-2.5 bg-primary text-white p-1 rounded-sm text-sm"
            type="submit"
          >
            <svg
              class="size-6"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.25 8.25 21 12m0 0-3.75 3.75M21 12H3"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
